<launch>
    
    <param name="/use_sim_time" value="ture"/>
    <node pkg="rosbag" type="play" name="playbag" args="--clock /home/ziv/sunshine_dateset/SR05.bag
         /velodyne_points:=/left/velodyne_points
         /rslidar_points:=/left/velodyne_points
         /ns1/rslidar_points:=/left/velodyne_points
         /ns2/rslidar_points:=/right/velodyne_points
         /ns1/velodyne_points:=/left/velodyne_points
         /ns2/velodyne_points:=/right/velodyne_points
         /points_raw:=/left/velodyne_points
    "/>

    <param name="file_save_path" type="string" value="/home/ziv/mloam/result/SR05/" />

    <!--    calibration part     -->
    <param name="need_calibra" type="bool" value="true" />
    <param name="manual_tz" type="bool" value="true" />
    <param name="tz_0_1" type="double" value="0" />
    <param name="require_loop" type="bool" value="true" />

    <!--    slam part     -->
    <param name="scan_line" type="int" value="16" />
    <param name="min_dis" type="double" value="0.2" />
    <param name="max_dis" type="double" value="100.0" />
    <param name="surf_filter_length" type="double" value="0.02"/>
    <param name="corn_filter_length" type="double" value="0.01"/>
    <param name="window_size" type="int" value="6" />
    <param name="margi_size" type="int" value="3" />
    <param name="keyframe_distance_threshold" type="double" value="0.9" />
    <param name="keyframe_angle_threshold" type="double" value="0.15" />
    <param name="save_map_resolution" type="double" value="0.05" />
    <param name="show_map_resolution" type="double" value="0.05" />
    <param name="gicp_fitness_socre" type="double" value="0.5" />
    <!--loop-->
    <param name="need_loop" type="bool" value="true" />
    <param name="loop_submap_length" type="int" value="5" />
    <param name="loop_detect_distance" type="int" value="1" />
    <param name="loop_found_sleep" type="int" value="5" />
    <param name="loop_skip_last" type="int" value="30" />

    <node pkg="mloam" type="laserProcessing" name="laserProcessing" output="screen" />

    <node pkg="mloam" type="laserOdometry" name="laserOdometry" output="screen" />

    <arg name="rviz" default="true" />
    <group if="$(arg rviz)">
        <node launch-prefix="nice" pkg="rviz" type="rviz" name="rviz" args="-d $(find mloam)/rviz/show.rviz" />
    </group>

</launch>
